### PR TITLE
DDCE-714: Section 4 navigation with updated index status functionality

### PR DESCRIPTION
**Pull request by robert-webb-hmrc**
# DDCE-714

##New feature

Verification of Section 4 journey including task list status update.

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/controllers/IndexController.scala [hunk 1]
```diff
@@ -43,21 +45,18 @@ class IndexController @Inject()(
         Spoke(charityInfoRoutes.CharityNameController.onPageLoad(NormalMode).url, getStatus(userAnswers.get(Section1Page)))))
 
       val section2 = TaskListSection(List(
```
Comment (abhijeetardale) [inline app/controllers/IndexController.scala:47]: Can we updated TaskListSectionas follows to get rid of unwanted Spoke List
TaskListSection(url: String, status: String)

File: app/views/Index.scala.html [hunk 2]
```diff
@@ -54,19 +54,19 @@ <h2 class="app-task-list__section">
             <ul class="app-task-list__items">
                 <li class="app-task-list__item">
                   <span class="app-task-list__task-name">
-                    <a href="@{controllers.regulatorsAndDocuments.routes.IsCharityRegulatorController.onPageLoad(NormalMode).url}"aria-describedby="name-regulator-completed">
+                    <a href="@{result.lift(1).get.spokes.head.url}"aria-describedby="name-regulator-completed">
```
Comment (abhijeetardale) [inline app/views/Index.scala.html:57]: with reference above we might need some refactoring here 
 result(1).url

File: app/views/Index.scala.html [hunk 3]
```diff
@@ -54,19 +54,19 @@ <h2 class="app-task-list__section">
...
                 </li>
                 <li class="app-task-list__item">
                   <span class="app-task-list__task-name">
-                    <a href="@{controllers.regulatorsAndDocuments.routes.SelectGoverningDocumentController.onPageLoad(NormalMode).url}"aria-describedby="governing-doc-info-completed">
+                    <a href="@{result.lift(2).get.spokes.head.url}"aria-describedby="governing-doc-info-completed">
                       @messages("index.section2.spoke2.label")
                     </a>
                   </span>
-                    <strong id="governing-doc-info-completed" class="govuk-tag govuk-tag--grey app-task-list__task-completed">@messages("index.section.notStarted")</strong>
+                    <strong id="governing-doc-info-completed" class='@if(result.lift(2).get.spokes.head.status.contains("completed")) {govuk-tag app-task-list__task-completed} else { govuk-tag govuk-tag--grey app-task-list__task-completed } '>@messages(result.lift(2).get.spokes.head.status)</strong>
```
Comment (abhijeetardale) [inline app/views/Index.scala.html:69]: same as above

**Comment by robert-webb-hmrc** [comment]
Updated with changes

**Pull request merged**